### ISSUE TITLE
add "recommended reading" sections to docs

### ISSUE THREAD

**Issue by techjeffharris** [nodejs/docs#100]
I would like to discuss the pros/cons of having a list of "recommended reading" items at the beginning of a higher-level reference (such as HTTP or Streams) that presents folks with concepts/APIs with which familiarity is recommended to better understand the current document.  HTTP, for example, is a high-level core-module that orchestrates many lower-level core modules: 
- `EventEmitter`
- `Buffer`
- `Streams`
- `net.Socket`
- `net.Server`
- etc...

So the "Recommended Reading" Section could be:
# HTTP

```
Stability: 2 - Stable
```
### API References
- [events](https://nodejs.org/api/events.html)
- [buffer](https://nodejs.org/api/buffer.html)
- [stream](https://nodejs.org/api/stream.html)
### Guides
- [Anatomy of an HTTP Transaction](https://github.com/nodejs/docs/blob/master/src/guides/anatomy-of-an-http-transaction.md)

**Comment by eljefedelrodeodeljefe** [comment]
👍  on the idea. Recommended reading within our docs/guides or external also?

**Comment by techjeffharris** [comment]
Definitely links to internal content, and perhaps links to reliable external sources such as MDN References.

**Comment by benjamingr** [comment]
I don't think it should be per module via "recommended reading". I think inlining them in the introductory section by making every new reference a link (a-la-wikipedia) would be really useful though.

**Comment by techjeffharris** [comment]
> I think inlining them in the introductory section

You would rather the references be integrated in plain English in the introduction, rather than in a list format, correct?

> making every new reference a link (a-la-wikipedia)

Sounds to me like you're proposing that we be more liberal with links to external sources like wikipedia, MDN, in addition to internal content, yes?

**Comment by Knighton910** [comment]
+1 from this guy.

**Comment by estliberitas** [comment]
Agree w/ @benjamingr - it should be some kind of introductory overview pointing people to exact documentation pages or sections in this pages, it makes more sense for anyone who is just starting developing with Node.

And agree about external links. Since we already do it in doc pages (links to man pages or general JS objects like `Error`), we can continue pointing people to stuff like MDN, linux.die.net, etc.

Talking about format, I'd make several paragraphs with links and after that provide a "Read more" list with link. So idea would be to join ideas of @techjeffharris and @benjamingr

**Comment by benjamingr** [comment]
@techjeffharris 

> Sounds to me like you're proposing that we be more liberal with links to external sources like wikipedia, MDN, in addition to internal content, yes?

No, I meant to use the same strategy used by wikipedia where terms are linked inline the first time they appear on the page.

**Comment by Trott** [comment]
Closing as this repository is dormant and likely to be archived soon. If this is still an issue, feel free to open it as an issue on the [main node repository](https://github.com/nodejs/node).
